x-pack/plugins/apm/public/components/shared/charts/transaction_coldstart_rate_chart/index.tsx

```diff
@@ -0,0 +1,184 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import {
+  EuiPanel,
+  EuiTitle,
+  EuiFlexGroup,
+  EuiFlexItem,
+  EuiIconTip,
+} from '@elastic/eui';
+import { i18n } from '@kbn/i18n';
+import React from 'react';
+import { APIReturnType } from '../../../../services/rest/createCallApmApi';
+import { asPercent } from '../../../../../common/utils/formatters';
+import { useFetcher } from '../../../../hooks/use_fetcher';
+import { useTheme } from '../../../../hooks/use_theme';
+import { TimeseriesChart } from '../timeseries_chart';
+import { useApmServiceContext } from '../../../../context/apm_service/use_apm_service_context';
+import {
+  getComparisonChartTheme,
+  getTimeRangeComparison,
+} from '../../time_comparison/get_time_range_comparison';
+import { useApmParams } from '../../../../hooks/use_apm_params';
+import { useTimeRange } from '../../../../hooks/use_time_range';
+import { TimeRangeComparisonType } from '../../../../../common/runtime_types/comparison_type_rt';
+
+function yLabelFormat(y?: number | null) {
+  return asPercent(y || 0, 1);
+}
+
+interface Props {
+  height?: number;
+  showAnnotations?: boolean;
+  kuery: string;
+  environment: string;
+  transactionName?: string;
+  comparisonEnabled?: boolean;
+  comparisonType?: TimeRangeComparisonType;
+}
+
+type ColdstartRate =
+  APIReturnType<'GET /internal/apm/services/{serviceName}/transactions/charts/coldstart_rate'>;
+
+const INITIAL_STATE: ColdstartRate = {
+  currentPeriod: {
+    transactionColdstartRate: [],
+    average: null,
+  },
+  previousPeriod: {
+    transactionColdstartRate: [],
+    average: null,
+  },
+};
+
+export function TransactionColdstartRateChart({
+  height,
+  showAnnotations = true,
+  environment,
+  kuery,
+  transactionName,
+  comparisonEnabled,
+  comparisonType,
+}: Props) {
+  const theme = useTheme();
+
+  const {
+    query: { rangeFrom, rangeTo },
+  } = useApmParams('/services/{serviceName}');
+
+  const { start, end } = useTimeRange({ rangeFrom, rangeTo });
+
+  const { serviceName, transactionType } = useApmServiceContext();
+  const comparisonChartThem = getComparisonChartTheme(theme);
+  const { comparisonStart, comparisonEnd } = getTimeRangeComparison({
+    start,
+    end,
+    comparisonType,
+    comparisonEnabled,
+  });
+
+  const endpoint = transactionName
+    ? ('GET /internal/apm/services/{serviceName}/transactions/charts/coldstart_rate_by_transaction_name' as const)
+    : ('GET /internal/apm/services/{serviceName}/transactions/charts/coldstart_rate' as const);
```

Comment:
I was expecting the data fetching part to be hoisted to the component that renders `TransactionColdstartRateChart`. I think having the data fetching _in_ the component kind of defeats the purpose of having two separate endpoints. It's kind of annoying that `TransactionCharts` is in between and I can see how it would be a big change. I'm OK with a follow-up ticket

Reply:
I agree , created this follow-up https://github.com/elastic/kibana/issues/124106